tpp/src/main/Member.js

```diff
@@ -265,30 +266,30 @@ export default class Member extends CoreMember {
      * @param currency - currency to redeem
      * @param description - optional transfer description
      * @param destinations - transfer destinations
-     * @param refId - ID that will be set on created Transfer
-     *                Token uses this to detect duplicates
-     *                caller might use this to recognize the transfer
-     *                if param empty, transfer will have random refId
+     * @param refId - ID that will be set on created Transfer.
+     *                Token uses this to detect duplicates.
+     *                Caller might use this to recognize the transfer.
+     *                If param empty, transfer will have random refId.
      * @return Transfer created as a result of this redeem call
      */
     redeemToken(
         token: Token | string,
-        amount?: number,
-        currency?: string,
-        description?: string,
-        destinations?: Array<TransferDestination> = [],
+        amount: ?number,
+        currency: ?string,
+        description: ?string,
+        destinations: Array<TransferDestination> = [],
         refId?: string
     ): Promise<Transfer> {
         return Util.callAsync(this.redeemToken, async () => {
             const finalToken = await this._resolveToken(token);
             if (!amount) {
-                amount = finalToken.payload.transfer.lifetimeAmount;
+                amount = finalToken && finalToken.payload.transfer.lifetimeAmount;
```

Comment:
do we need this check here?

Reply:
i reverted, don't remember making these changes